File: baekjoon/simulation/9517/Main.java
import java.util.*;

class Main {
    
    public static final int timeover = 210;
    
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int time = 0;
        int first = sc.nextInt();
        int n = sc.nextInt();
        for(int i=0; i<n; i++) {
            int a = sc.nextInt();
            char b = sc.next().charAt(0);
            time += a;
            if(time >= timeover) {
                System.out.println(first);
                return ;
            }
            if(b == 'T') first = (first+1==9 ? 1 : first+1);
        }
    }
}

File: baekjoon/simulation/9517/main.cpp
#include <iostream>
#include <cstdio>
#include <vector>

using namespace std;
const int timeover = 210;

int main()
{
    int first;
    int n;
    scanf("%d %d", &first, &n);
    first--;
    int time = 0;
    for(int i=0; i<n; i++) {
        int a;
        char b;
        scanf("%d %c", &a, &b);
        bool boom = false;
        time+=a;
        if(time >= timeover) {
            printf("%d\n", first+1);
            return 0;
        }
        if(b == 'T') first = (first+1==8 ? 0 : first+1);
    }
    
    return 0;
}